mirgecom/fluid.py

```diff
@@ -391,10 +390,10 @@ def velocity_gradient(discr, cv, grad_cv):
         \partial_{x}\mathbf{v}_{y}&\partial_{y}\mathbf{v}_{y} \end{array} \right)$
 
     """
-    velocity = cv.momentum / cv.mass
-    return (1/cv.mass)*make_obj_array([grad_cv.momentum[i]
-                                       - velocity[i]*grad_cv.mass
-                                       for i in range(cv.dim)])
+    vel = cv.velocity
+    return (1/cv.mass)*np.array([grad_cv.momentum[i]
+                                 - vel[i]*grad_cv.mass
+                                 for i in range(cv.dim)], dtype=object)
```

Comment:
`np.array(..., dtype=object)` can have some pretty perplexing behavior (e.g., sometimes it traverses `DOFArray`s and converts them to numpy arrays, and sometimes it doesn't; I still haven't figured out how it decides), so I'm tempted to say it's best avoided altogether.

I think this can be written as

```suggestion
    return (grad_cv.momentum - np.outer(cv.velocity, grad_cv.mass))/cv.mass
```
right?

Reply:
Yikes! Thanks for rescuing this, then.  Let's try it out.